### PR TITLE
fix(mac): load only 80 characters from context when processing keystrokes

### LINKED ISSUE #10970: bug(mac): Southern Carrier keyboard producing incorrect output after typing several sentences
### Describe the bug

I was testing out the new fv_southern_carrier keyboard using Keyman 17.0.285-beta and the keyboard stopped working as expected. 

The sequence I typed was <kbd>t</kbd><kbd>s</kbd><kbd>'</kbd><kbd>i</kbd><kbd>y</kbd><kbd>a</kbd><kbd>w</kbd><kbd>h</kbd>

I expected `ᙫᘓᐁᑋ`
I received `ᙫyᐊwᑋ`
Another time, I received `ᐪᔆ’ᐉyᐊwᑋ`

Everything typed afterwards outputs either the latin value or the final form of each syllable. I attempted using the dead-key that is on the keyboard to separate the syllables but the error continued to occur. I tried typing the same sequence in different programs and had the same result each time (Microsoft Teams, Discord, Slack, Microsoft Word).

You may need to type a few sentences prior to this issue being reproducible. If needed, I can provide some sample text to type.

### Reproduce the bug

_No response_

### Expected behavior

`ᙫᘓᐁᑋ`

### Related issues

_No response_

### Keyman apps

- [ ] Keyman for Android
- [ ] Keyman for iPhone and iPad
- [ ] Keyman for Linux
- [X] Keyman for macOS
- [ ] Keyman for Windows
- [ ] Keyman Developer
- [ ] KeymanWeb
- [ ] Other - give details at bottom of form

### Keyman version

17.0.285-beta

### Operating system

MacOS Sonoma 14.4

### Device

iMac

### Target application

Microsoft Teams, Discord, Slack, Microsoft Word

### Browser

_No response_

### Keyboard name

fv_southern_carrier

### Keyboard version

_No response_

### Language name

Southern Carrier

### Additional context

_No response_

### LINKED ISSUE #11087: bug(mac): typing in Pages is painfully slow
### Describe the bug

Using Pages, typing with Keyman was slow and seemed to get slower. 

Typing in TextEdit seemed responsive -- rendering the text was able to keep pace with the typing.

### Reproduce the bug

1. Open a new document in Pages
2. Type quickly

### Expected behavior

_No response_

### Related issues

_No response_

### Keyman apps

- [ ] Keyman for Android
- [ ] Keyman for iPhone and iPad
- [ ] Keyman for Linux
- [X] Keyman for macOS
- [ ] Keyman for Windows
- [ ] Keyman Developer
- [ ] KeymanWeb
- [ ] Other - give details at bottom of form

### Keyman version

18.0.295

### Operating system

macOS Ventura

### Device

MacBook Pro with Intel

### Target application

Apple Pages

### Browser

_No response_

### Keyboard name

_No response_

### Keyboard version

_No response_

### Language name

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sgschantz**
When reading the current context to process keystrokes, load only 80 characters instead of the entire context.

Asking for all of the existing context degrades performance in Pages and appears to cause null to be returned when exceeding a certain limit in Slack.

Fixes #11087
Fixes #10970 

# User Testing

* **TEST_TYPING_IN_PAGES**: typing reasonable amounts of text should not become noticeable slow

1. Open Pages and create a new document
1. Switch to the EuroLatin (SIL) keyboard
1. Type a couple hundred characters
1. Confirm that inserting characters does not become noticeably slower as you continue to type (for comparison try typing with the Apple US keyboard and see if it is equally responsive)

* **TEST_TYPING_IN_SLACK**: keys should continue processing correctly after typing over 100 characters

1. Open Slack
1. Switch to the fv_southern_carrier keyboard
1. Go to compose a new message to yourself and type at least 101 characters
1. Without moving from the text field where you are typing, type the following sequence of keys: <kbd>t</kbd><kbd>s</kbd><kbd>'</kbd><kbd>i</kbd><kbd>y</kbd><kbd>a</kbd><kbd>w</kbd><kbd>h</kbd>
1. Confirm that the resulting text for these additional keystrokes is ᙫᘓᗛ

### CODE REVIEW

File: mac/Keyman4MacIM/Keyman4MacIM/KMInputMethodEventHandler.m [hunk 1]
```diff
@@ -306,21 +306,33 @@ -(void)reportContext:(NSEvent *)event forClient:(id) client {
 }
 
 -(NSString*)readContext:(NSEvent *)event forClient:(id) client {
-  NSString *contextString = nil;
+  NSString *contextString = @"";
   NSAttributedString *attributedString = nil;
   
-  // if we can read the text, then get the context
+  // if we can read the text, then get the context for up to kMaxContent characters
```
Comment (darcywong00) [inline mac/Keyman4MacIM/Keyman4MacIM/KMInputMethodEventHandler.m:312]:
```suggestion
  // if we can read the text, then get the context for up to kMaxContext characters
```

**Review by mcdurdin — APPROVED** [review]
LGTM. Not only fixing the bug but targeted tidying up of the existing code also. I like.

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_TYPING_IN_PAGES (PASSED):** Tested with the attached PR build (Keyman 17.0.300-beta-test-11141) on macOS Ventura and here is my observation: 1. Installed EuroLatin (SIL) keyboard. 2. Opened Pages application. 3. Typed a paragraph (exceeding 700 characters) and confirmed that there was no noticeable delay when inserting characters into the document. Everything appears to be functioning properly.

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_TYPING_IN_SLACK (PASSED):** 1. Installed fv_southern_carrier keyboard. 2. Opened Slack. 3. In the message box, typed more than 101 characters. Without moving from the text field, I typed the following sequence of keys : ts'iyawh. 4. Verified that the resulting text for these additional keystrokes displayed  ᙫᘓᗛ  in the field. Seems to be working fine.

**Pull request merged**